scripts/codeCoverage.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+LAUNCH_DIR=`pwd`
+
+#  uncoment to enabled source blacklist: exmple --> SOURCE_BLACKLIST=".*ProbeTransmogrifier.cpp.*"
+HEADER_WHITELIST=
+
+
+
+cd 3rdparty
+unzip -u gtest-1.7.0.zip
+cd ..
+
+#clean up coverage dir
+COVERAGE_DIR=coverage
+rm -rf $COVERAGE_DIR
+mkdir -p $COVERAGE_DIR
+
+rm -rf build
+mkdir -p build
+cd build
+
+
+
+# A dummy version to please cmake
+VERSION=1.1.1.1.1 
+echo "Pseudo FileIO version: $VERSION"
+PATH=/usr/local/probe/bin:$PATH
+/usr/local/probe/bin/cmake -DUSE_LR_DEBUG=ON -DVERSION=$VERSION -DCMAKE_CXX_COMPILER_ARG1:STRING=' -Wall -Werror -g -gdwarf-2 -fprofile-arcs -ftest-coverage -O0 -fPIC -m64 -Wl,-rpath -Wl,. -Wl,-rpath -Wl,/usr/local/probe/lib -Wl,-rpath -Wl,/usr/local/probe/lib64 ' -DCMAKE_CXX_COMPILER=/usr/local/probe/bin/g++ ..
+
+
+make -j
+./UnitTestRunner --gtest_filter=-*Root
+
+#run the root tests once as non-root. They will fail but will generate the .gcda files
+./UnitTestRunner --gtest_filter=*Root 
+
+# the bug in the code coverage for root has to do with the under the cover
+# root manipulation some FileIO code does. Now when the .gcda
+# files already exist we can execute the test again as root
+sudo ./UnitTestRunner --gtest_filter=*Root
```

Comment:
I wasn't sure about this one. 

-*Root:   Will collect all the coverage for the unit tests except the Root ones
*Root:   Will collect the coverage for the Root unit tests. It will also **fail** the tests since it is not run as root. 

sudo ... *Root: Will not collect any coverage for the Root unit tests but run them without failure (due to privileges)